### ISSUE TITLE
VSC: TypeScript types aren't refreshed when changing a document type definition

### ISSUE THREAD

**Issue by schickling** [contentlayerdev/contentlayer#65]
Unfortunately in some situations e.g. when changing the document type definitions which causes the TS type definitions to be regenerated, the TS LSP in VSC doesn't automatically pick up those changes. [This seems to be a general problem with TS and VSC](https://github.com/microsoft/vscode/issues/88315).

For now users need to manually [`Restart TS server`](https://stackoverflow.com/a/64456017) but we could also investigate other solutions to this problem similar to [what the Prisma VSC extension is doing](https://github.com/prisma/language-tools/blob/9d042a3ad4431985c07f5f789be1fff53db6677e/packages/language-server/src/server.ts#L207).

![](https://i.stack.imgur.com/zRhGr.png)

**Comment by schickling** [comment]
In order to work around this issue I'm currently also exploring a different "file structure" that generates the files into a `<root>/.contentlayer` folder (instead of `<root>/node_modules/.contentlayer`) which would also require the following addition to a project's `tsconfig.json`:

```
    "paths": {
      "contentlayer/*": ["./.contentlayer/*"]
    }
```

![CleanShot 2021-12-08 at 16 42 46](https://user-images.githubusercontent.com/1567498/145237978-a3649135-ee4b-4788-b1eb-cec7862ed5be.png)


Downside: For JS-only projects this requires a `jsconfig.json` or `tsconfig.json` (see [Next.js paths docs](https://nextjs.org/docs/advanced-features/module-path-aliases)).

**Comment by seancdavis** [comment]
TLDR: I strongly support this adjustment, while noting that `tsconfig.json`/`jsconfig.json` is a _convenience_ not a requirement.

---

In fact, I've already been using a symlink from `node_modules/.contentlayer` to `.contentlayer` to make it easier to find and inspect these generated files.

As a plus, `tsconfig.json`/`jsconfig.json` is a convenience to avoid working with relative imports. Given that, and in thinking about documentation, I don't think we should assume their existence in projects or in the docs we're writing. We can (and _should_) recommend them as improving DX, just not assume they exist.

**Comment by schickling** [comment]
This problem should now be addressed with [version `0.1.0`](https://github.com/contentlayerdev/contentlayer/releases/tag/v0.1.0). If you're running into problems, please let me know. :)
